src/EntryPoints/PersistentPageIdFunction.php

```diff
@@ -29,7 +29,7 @@ public function handleParserFunctionCall( Parser $parser ): array {
 		}
 
 		return [
-			$this->idFormatter->format( $this->getPersistentIdForPage( $page ) ),
+			htmlspecialchars( $this->idFormatter->format( $this->getPersistentIdForPage( $page ) ) ),
```

Comment:
Is HTML escaping needed/appropriate here? Note the below `'isHTML' => false,`.

Reply:
None of those flags make a difference to the escaping of HTML:
https://www.mediawiki.org/wiki/Manual:Parser_functions#The_setFunctionHook_hook

It say `isHtml` is about making the value "armoured against wikitext transformation". anyhow, that doesn't escape it either way.

I think that's more a case of telling MW that the desired output is going to be actual HTML and that it should handle it in a specific way.